### PR TITLE
rewards endpoint changes

### PR DESCRIPTION
**Pull request by eesmerdag**
https://www.notion.so/figmentnetworks/account-check-for-rewards-endpoint-49f8b7c8dc5a486f9c71bdc0b8d34748

### CODE REVIEW

File: store/delegators.go [hunk 1]
```diff
@@ -18,18 +19,58 @@ type DelegatorsStore struct {
...
 func (s *DelegatorsStore) FetchRewardsByInterval(account string, validatorId string, from time.Time, to time.Time, timeInterval model.TimeInterval) ([]model.RewardsSummary, error) {
+	q := "SELECT to_char(distributed_at_time, $INTERVAL) AS interval, validator_id as validator, SUM(reward) AS amount FROM delegator_epochs "
+
+	var (
+		args   []interface{}
+		wherec []string
+		i      = 1
+	)
+
+	if account != "" {
```
Comment (sosedoff) [inline store/delegators.go:30]: I would prefer if we are not building the full query via string concatenation. You can use gorm for this pretty easily (similar to how we use scopes in other places).

File: server/params.go [hunk 2]
```diff
@@ -57,8 +57,8 @@ func (p *statsParams) Validate() error {
 }
 
 type rewardsParams struct {
-	From     time.Time `form:"from" binding:"required" time_format:"2006-01-02"`
-	To       time.Time `form:"to" binding:"required" time_format:"2006-01-02"`
+	From     time.Time `form:"from" binding:"-" time_format:"2006-01-02"`
```
Comment (sosedoff) [inline server/params.go:60]: dont think you need `binding:"-"` here at all.

**Review by sosedoff — CHANGES_REQUESTED** [review]
Looking good so far - i left a few comments, would be great if we're not building SQL by hand.

File: store/delegators.go (hunk 1 above)
Comment (eesmerdag) [reply store/delegators.go:30]: 👍

File: store/delegators.go [hunk 3]
```diff
@@ -18,18 +18,39 @@ type DelegatorsStore struct {
...
-	var res []model.RewardsSummary
-	q := strings.Replace(queries.DelegatorsRewards, "$INTERVAL", "'"+timeInterval.String()+"'", -1)
-	var err error
-	if validatorId == "" {
-		q = strings.Replace(q, "AND validator_id = ?", "", -1)
-		err = s.db.Raw(q, account, from, to).Scan(&res).Error
-	} else {
-		err = s.db.Raw(q, account, validatorId, from, to).Scan(&res).Error
+	slt := " to_char(distributed_at_time, $INTERVAL) AS interval, validator_id as validator, SUM(reward) AS amount"
+	slt = strings.Replace(slt, "$INTERVAL", "'"+timeInterval.String()+"'", -1)
```
Comment (sosedoff) [inline store/delegators.go:22]: i was hoping we can use named arguments: https://gorm.io/docs/sql_builder.html#Named-Argument - does not seem to be working, maybe its only supported by the newer versions of gorm?

**Pull request merged**